src/metatrain/experimental/mace/model.py

```diff
@@ -0,0 +1,658 @@
+        self.requested_nl = NeighborListOptions(
+            cutoff=self.cutoff,
+            full_list=True,
+            strict=True,
+        )
+
+        # ---------------------------
+        #   Store info about MACE
+        # ---------------------------
+
+        # Atomic species information
+        self.atomic_types = self.mace_model.atomic_numbers.tolist()
+        self.register_buffer(
+            "atomic_types_to_species_index",
+            torch.zeros(max(self.atomic_types) + 1, dtype=torch.int64),
+        )
+        for i, atomic_type in enumerate(self.atomic_types):
+            self.atomic_types_to_species_index[atomic_type] = i
+
+        # Information about the irreps of MACE features
+        self.per_layer_irreps = [
+            product.linear.irreps_out for product in self.mace_model.products
+        ]
+        self.features_irreps = sum(self.per_layer_irreps, o3.Irreps())
+
+        # ---------------------------
+        #    Add heads for targets
+        # ---------------------------
+
+        # Create heads for each target, store the target info for each of them.
+        self.heads = torch.nn.ModuleDict()
+        self.target_infos: Dict[str, TargetInfo] = {}
+        for target_name, target_info in dataset_info.targets.items():
+            self._add_output(target_name, target_info)
+
+        self.target_infos["features"] = get_e3nn_target_info(
+            "features",
+            {
+                "type": {"spherical": {"irreps": self.features_irreps}},
+                "per_atom": True,
+                "properties_name": "feature",
+            },
+        )
+
+        self.outputs = {
+            k: ModelOutput(
+                quantity=target_info.quantity,
+                unit=target_info.unit,
+                per_atom=True,
+            )
+            for k, target_info in self.target_infos.items()
+        }
+
+        # ---------------------------
+        # Data preprocessing modules
+        # ---------------------------
+
+        # The composition model and scaler are handled by the trainer during training.
+        # Their purpose is to adapt the data for optimal training.
+        # At evaluation time, the model applies them on forward.
+        composition_model = CompositionModel(
+            hypers={},
+            dataset_info=DatasetInfo(
+                length_unit=dataset_info.length_unit,
+                atomic_types=self.atomic_types,
+                targets={
+                    target_name: target_info
+                    for target_name, target_info in dataset_info.targets.items()
+                    if CompositionModel.is_valid_target(target_name, target_info)
+                },
+            ),
+        )
+        self.additive_models = torch.nn.ModuleList([composition_model])
+
+        self.scaler = Scaler(hypers={}, dataset_info=dataset_info)
+
+        self.finetune_config: Dict[str, Any] = {}
+
+    def restart(self, dataset_info: DatasetInfo) -> "MetaMACE":
```

Comment:
```suggestion
    def restart(self, dataset_info: DatasetInfo) -> Self:
```

or is this forbidden by torch script.

Reply:
Is metatrain python >= 3.11?